cmd/soci/commands/ztoc/list.go

```diff
@@ -82,52 +76,58 @@ var listCommand = &cli.Command{
 			}
 			defer cancel()
 
+			cs := client.ContentStore()
 			is := client.ImageService()
 			img, err := is.Get(ctx, imgRef)
 			if err != nil {
 				return err
 			}
-			platform, err := images.Platforms(ctx, client.ContentStore(), img.Target)
+
+			supportedPlatforms, err := images.Platforms(ctx, client.ContentStore(), img.Target)
 			if err != nil {
-				return err
+				return fmt.Errorf("failed to get supported platforms for the image: %w", err)
 			}
-			var layers []ocispec.Descriptor
-			for _, p := range platform {
-				manifest, err := images.Manifest(ctx, client.ContentStore(), img.Target, platforms.OnlyStrict(p))
-				if err != nil && verbose {
-					// print a warning message if a manifest can't be resolved
-					// continue looking for manifests of other platforms
-					fmt.Printf("no image manifest for platform %s/%s. err: %v\n", p.Architecture, p.OS, err)
-				} else {
-					layers = append(layers, manifest.Layers...)
+			indexInfos, _, err := soci.GetIndexDescriptorCollection(ctx, cs, db, img, supportedPlatforms)
+			if err != nil {
+				return fmt.Errorf("failed to get soci indexes for the image: %w", err)
+			}
+
+			var ztocDescs []ocispec.Descriptor
+			for _, indexInfo := range indexInfos {
+				readerAt, err := cs.ReaderAt(ctx, indexInfo.Descriptor)
+				if err != nil {
+					return fmt.Errorf("failed to read a soci index with digest %s from the content store: %w", indexInfo.Descriptor.Digest, err)
+				}
+				var sociIndex soci.Index
+				err = soci.DecodeIndex(io.NewSectionReader(readerAt, 0, indexInfo.Descriptor.Size), &sociIndex)
+				if err != nil {
+					return fmt.Errorf("failed to decode a soci index with digest %s: %w", indexInfo.Descriptor.Digest, err)
+				}
+				for _, blob := range sociIndex.Blobs {
+					if blob.MediaType != soci.SociLayerMediaType {
+						continue
+					}
+					ztocDescs = append(ztocDescs, blob)
 				}
 			}
-			if len(layers) == 0 {
-				return fmt.Errorf("no image layers. could not filter ztoc")
+			if len(ztocDescs) == 0 {
```

Comment:
With the old behavior, no indexes/ztocs for an image was not considered an error, but now it is. Is that what we want?